lib/stackops/apiGateway.js

```diff
@@ -267,7 +268,7 @@ module.exports = function(currentTemplate, aliasStackTemplates, currentAliasStac
 			const aliasName = _.find(_.keys(aliases), alias => _.startsWith(alias, functionName));
 
 			// Adjust references and alias permissions
-			permission.Properties.FunctionName = { Ref: aliasName };
+			permission.Properties.FunctionName = aliasName ? { Ref: aliasName } : permission.Properties.FunctionName;
```

Comment:
I don't think this change is not an appropriate fix. It will affect _all_ permission resources and unconditionally use any existing function name in case something is wrong with the alias.

Using the original function name must only be done if we are really sure that the target is an external function, and not a function deployed by the current service. If that happens for a local function this leads to untraceable errors and lets the permissions be attached to $LATEST instead of the alias silently.

We should first check if the permission targets a service external function and handle that in a **separate code branch**. This allows for handling configuration errors differently for internal functions and external ones instead of mixing them and swallowing errors.

The _.compact() change below is also affected. It must only be in place for external functions, not internal ones. If permissions that reference internal functions are not dependent on an alias, there is a problem with the service.